addons/mail/models/mail_message.py

```diff
@@ -902,7 +902,7 @@ def _record_by_message(self):
             message: self.env[message.model]
             .browse(message.res_id)
             .with_prefetch(records_by_model_name[message.model]._prefetch_ids)
-            for message in self.filtered(lambda m: m.model and m.res_id)
+            for message in self.filtered(lambda m: m.model and m.res_id) if self.env[message.model].browse(message.res_id).exists()
```

Comment:
They should already be filtered in `_records_by_model_name`
Rather than re-doing the exists query, we should check that `message.res_id` is in `records_by_model_name[message.model]`